superset/datasets/schemas.py

```diff
@@ -145,6 +146,34 @@ class DatasetRelatedObjectsResponse(Schema):
     dashboards = fields.Nested(DatasetRelatedDashboards)
 
 
+class DatasetResponse(Schema):
+    id = fields.Int()
+    cache_timeout = fields.Int()
+    columns = fields.List(fields.Dict())
```

Comment:
Setting this as `fields.List(fields.Dict())` causes it to lose the datatype. This is after this change:
<img width="922" alt="image" src="https://user-images.githubusercontent.com/33317356/232698854-769e8dc0-a3f8-4fd5-bbc9-322cf2f94ced.png">

This is before the change:
<img width="919" alt="image" src="https://user-images.githubusercontent.com/33317356/232699232-dbaab196-9936-4196-921c-da0200eed33b.png">

Here's a schema that I believe is more complete that could probably serve as a better template:
https://github.com/apache/superset/blob/c197bf9e6db85a76d8118a4ec11a83ca2f6aad6d/superset/datasets/schemas.py#L47-L63